src/viewport/s3_service.py

```
@@ -441,16 +494,24 @@ async def delete_folder(self, prefix: str) -> None:
                             if "Errors" in response:
                                 for error in response["Errors"]:
-                                    logger.warning("Failed to delete object %s: %s", error["Key"], error["Message"])
-                        except Exception:
-                            logger.exception("Failed to delete batch of objects")
+                                    logger.warning(
+                                        "Failed to delete object key_hash=%s: code=%s",
+                                        _key_hash(str(error.get("Key"))),
+                                        error.get("Code"),
+                                    )
+                                    record_s3_operation("delete_batch", "partial")
+                        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)